service/src/main/java/bio/terra/drshub/services/DrsResolutionService.java

```diff
@@ -257,13 +260,16 @@ AccessURL fetchDrsObjectAccessUrl(
       TypeEnum accessMethodType,
       List<DrsHubAuthorization> drsHubAuthorizations,
       AuditLogEvent.Builder auditLogEventBuilder,
+      String ip,
       String googleProject) {
 
     var drsApi = drsApiFactory.getApiFromUriComponents(uriComponents, drsProvider);
     var objectId = getObjectId(uriComponents);
+    drsApi.setHeader("x-forwarded-for", ip);
```

Comment:
I didn't limit this to TDR requests only since it's a header and not part of the response

Reply:
Do you know what happens if the `ip` is null? Maybe we should only set it as a header if it's defined

Reply:
yeah that's a good point, I'll add a check for that